app/services/alerts/collate_benchmark_data.rb

```diff
@@ -23,14 +23,15 @@ def get_benchmarks_for_latest_run(latest_school_runs, benchmarks)
       latest_school_runs.each do |benchmark_result_school_generation_run|
         school_id = benchmark_result_school_generation_run.school_id
 
-        benchmark_result_school_generation_run.benchmark_results.pluck(:asof, result_column).each do |benchmark_result|
-          unless benchmarks.key?(benchmark_result[0])
-            benchmarks[benchmark_result[0]] = { school_id => {} }
+        benchmark_result_school_generation_run.benchmark_results.pluck(result_column).each do |benchmark_result|
+          # When collating results we need to collate around benchmark run date (created_at) not analysis (asof) date
+          unless benchmarks.key?(@benchmark_run.created_at)
```

Comment:
This should be `@benchmark_run.run_date` throughout, in order to be consistent with `benchmarks_controller`.

Maybe assign the date to a var in the method then reuse through?